src/retryLink.ts

```diff
@@ -0,0 +1,66 @@
+import {
+  Operation,
+  NextLink,
+  FetchResult,
+} from './types';
+import * as Observable from 'zen-observable';
+import {
+  ensureForward,
+} from './linkUtils';
+
+import {
+  ApolloLink,
+} from './link';
+
+export default class RetryLink extends ApolloLink {
+
+  private count: number = 0;
+  private delay: number;
+  private max: number;
+  private interval: (delay: number, count: number) => number;
+
+  constructor (params?: {
+    max?: number,
+    delay?: number,
+    interval?: (delay: number, count: number) => number,
+  }) {
+    super();
+    this.max = params && params.max || 10;
+    this.delay =  params && params.delay || 300;
+    this.interval =  params && params.interval || this.defaultInterval;
+  }
+
+  public request(operation: Operation, forward: NextLink): Observable<FetchResult> {
+    ensureForward(forward);
```

Comment:
should this be attached to the Link interface so it can be used without importing? The forward need is still I think the confusingish part here, so lets try to make it an easy thing for library developers to use